### PR TITLE
Docs: ArrayedCollections / SequencableCollections clarified

### PR DESCRIPTION
**Pull request by miriamvoth**
Clarified documentation regarding element checks for equality / identity

## Purpose and Motivation

Connected to #4573, but not completely fixing it yet.

## Types of changes

- Documentation

## To-do list

- [x] Code is tested
- [x] All tests are passing
- [x] Updated documentation
- [x] This PR is ready for review

### CODE REVIEW

File: HelpSource/Classes/ArrayedCollection.schelp [hunk 1]
```diff
@@ -465,14 +465,16 @@ a.reshape(2,3);
 ::
 
 method::find
-Finds the starting index of a number of elements contained in the array.
+Finds the starting index of a number of elements contained in the array. This method expects a collection as an argument. So for finding only one element, place the element inside a collection.
```
Comment (telephon) [inline HelpSource/Classes/ArrayedCollection.schelp:468]: Wouldn't it be better to recommend `indexOfEqual` when one searches only for one element? At least it is more efficient, I think.

File: HelpSource/Classes/SequenceableCollection.schelp [hunk 2]
```diff
@@ -175,7 +175,7 @@ copymethod:: Collection -minIndex
 
 
 method::find
-If the strong::sublist:: exists in the receiver (in the specified order), at an offset greater than or equal to the initial strong::offset::, then return the starting index.
+If the strong::sublist:: exists in the receiver (in the specified order), at an offset greater than or equal to the initial strong::offset::, then return the starting index. The type of the sublist needs to match the type of the list to search in. Elements are checked for equality (not for identity).
```
Comment (telephon) [inline HelpSource/Classes/SequenceableCollection.schelp:178]: Better not talk of "types", but of "class".

File: HelpSource/Classes/ArrayedCollection.schelp (hunk 1 above)
Comment (miriamvoth) [reply HelpSource/Classes/ArrayedCollection.schelp:468]: Oh, yes thats a very good point. I did not think about this.

File: HelpSource/Classes/SequenceableCollection.schelp (hunk 2 above)
Comment (miriamvoth) [reply HelpSource/Classes/SequenceableCollection.schelp:178]: Thanks for pointing that out. I did not know about the different meaning of type and class. 
"Same kind" sounds very generic to me though. I would propose  "The sublist must be of the same kind (class) as the list to search in."

Comment (telephon) [reply HelpSource/Classes/SequenceableCollection.schelp]: yes, I agree, that's good.

**Comment by dyfer** [comment]
@telephon are OK with these changes? And if so, would you approve the PR?

**Review by telephon — APPROVED** [review]

**Comment by dyfer** [comment]
thanks @miriamvoth !

**Pull request merged**